### PR TITLE
ui: More transit flakiness

### PR DESCRIPTION
**Pull request by hellobontempo**
The refactor I did previously helped determine that these tests fail randomly - the screenshots show the last few failures and a different key type failed each time. 
Screenshot 2023-09-11 at 7 10 44 PM

It looks like the rotate action times out occasionally, perhaps due to a race condition in the test runner. When the test failed locally there were no console errors or anything indicating an actual probably occurred.

I'm not sure why so many key combinations were originally tested, so I've reverted some of the changes in this PR https://github.com/hashicorp/vault/pull/22617 and kept those tests (skipped) so when we make future improvements to transit we can run them locally and make updates accordingly.

### CODE REVIEW

File: ui/tests/acceptance/transit-test.js [hunk 1]
```diff
@@ -17,134 +17,68 @@ const SELECTORS = {
...
+  form: (item) => `[data-test-transit-key="${item}"]`,
+  versionRow: (version) => `[data-test-transit-version="${version}"]`,
   rotate: {
     trigger: '[data-test-confirm-action-trigger]',
     confirm: '[data-test-confirm-button]',
   },
 };
 
-// convergent
-const groupOne = [
```
Comment (hellobontempo) [inline ui/tests/acceptance/transit-test.js:28]: I kept these combinations for posterity below for use in the skipped tests

File: ui/tests/acceptance/transit-test.js [hunk 2]
```diff
@@ -17,134 +17,68 @@ const SELECTORS = {
...
+const KEY_TYPES = [
```
Comment (hellobontempo) [inline ui/tests/acceptance/transit-test.js:31]: these are simplified key types to test the form view updates appropriately

File: ui/tests/acceptance/transit-test.js [hunk 3]
```diff
@@ -17,134 +17,68 @@ const SELECTORS = {
...
-    supportsEncryption: true,
-    autoRotate: true,
+    type: 'chacha20-poly1305',
+    'convergent-encryption': true,
+    derived: true,
+    exportable: true,
   },
 ];
 
-const generateTransitKey = async function (key, uid) {
```
Comment (hellobontempo) [inline ui/tests/acceptance/transit-test.js:122]: using the browser console instead to speed up these tests

File: ui/tests/acceptance/transit-test.js [hunk 4]
```diff
@@ -310,16 +192,28 @@ const testConvergentEncryption = async function (assert, keyName) {
...
+      const config = [];
+
+      if (key.exportable) config.push('exportable=true');
+      if (key.derived) config.push('derived=true');
+      if (key.convergent) config.push('convergent_encryption=true');
+      if (key.autoRotate) config.push('auto_rotate_period=720h');
+
+      await runCmd([`vault write ${this.path}/keys/${name} type=${key.type} ${config.join(' ')} -f`]);
+      return name;
+    };
```
Comment (zofskeez) [inline ui/tests/acceptance/transit-test.js:216]: 🎉

**Review by zofskeez — APPROVED** [review]
It feels a bit like whack-a-mole with some of these flaky tests! Between the updates and the skips hopefully this will stabalize the transit tests until we can revisit. 🎉

**Pull request merged**